subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -161,4 +170,34 @@ public boolean isSatisfiedBy(Task element) {
 
         // TODO - check should depend on test
     }
+
+    private void configureTestedComponent(final Project project) {
+        project.getPlugins().withType(SwiftExecutablePlugin.class, new Action<SwiftExecutablePlugin>() {
+            @Override
+            public void execute(SwiftExecutablePlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+
+        project.getPlugins().withType(SwiftLibraryPlugin.class, new Action<SwiftLibraryPlugin>() {
+            @Override
+            public void execute(SwiftLibraryPlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+    }
+
+    private void configureTestedSwiftComponent(Project project) {
+        TaskContainer tasks = project.getTasks();
+        DirectoryVar buildDirectory = project.getLayout().getBuildDirectory();
+
+        SwiftCompile compile = tasks.withType(SwiftCompile.class).getByName("compileSwift");
+
+        SwiftCompile compileTest = tasks.withType(SwiftCompile.class).getByName("compileTestSwift");
+        compileTest.dependsOn(compile);
+        compileTest.includes(buildDirectory.dir("main/objs"));
+
+        AbstractLinkTask linkTest = tasks.withType(AbstractLinkTask.class).getByName("linkTest");
+        linkTest.source(compile.getObjectFileDirectory().getAsFileTree().matching(new PatternSet().include("**/*.obj", "**/*.o").exclude("**/main.o")));
```

Comment:
With other native languages that would be true, however, it seems that Swift use a hard coded convention that explicitly specifies [`main.swift` as been the entry point for executables](https://github.com/apple/swift-package-manager/blob/57c5be1db1c1e12e089dff02241ffbce5722fb0e/Documentation/PackageManagerCommunityProposal.md#L78). If we don't exclude, we can't run XCTest on executable source code. I changed it to configure the test case instead. However, I rely on the implementation of `AbstractLinkTask`.